src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -880,7 +880,7 @@ private static void updateDependencyVersions(Element element, ModelBase model) {
     private static void updateDependencyVersions(Element dependenciesElement, List<Dependency> dependencies) {
         final Map<String, String> dependencyVersionMap = dependencies.stream()
                 .filter(it -> it.getVersion() != null)
-                .collect(toMap(it -> it.getGroupId() + ":" + it.getArtifactId(), Dependency::getVersion));
+                .collect(toMap(Dependency::getManagementKey, Dependency::getVersion));
```

Comment:
I think we need to adjust line 890 as well, don't we?

Reply:
Changing only line 883 fixed the issue for me, so I didn't really look more into it, but now that I do, it seems like a better fix yes.
So something like
```
String dependencyGroupId = dependencyElement.getChild("groupId").getText();
String dependencyArtifactId = dependencyElement.getChild("artifactId").getText();
String dependencyType = dependencyElement.getChild("type").getText();
String dependencyClassifier = dependencyElement.getChild("classifier") != null ? dependencyElement.getChild("classifier").getText() : null;
Element dependencyVersionElement = dependencyElement.getChild("version");
if (dependencyVersionElement != null) {
    dependencyVersionElement.setText(dependencyVersionMap.get(dependencyGroupId + ":" +
                                                              dependencyArtifactId + ":" +
                                                              dependencyType +
                                                              (dependencyClassifier != null ? ":" + dependencyClassifier : "")));
```
to mimic the getManagementKey method ?

Reply:
Here is a test project for this bug.
https://github.com/Bimeche/TestExtension